code/modules/projectiles/gun.dm

```diff
@@ -309,6 +309,14 @@
 
 	return !P.launch_from_gun(target, user, src, target_zone, x_offset, y_offset)
 
+//aprt of reskins that have two sprites, no touch please
+/obj/item/weapon/gun/projectile/colt/detective/attack_hand(var/mob/living/user)

```

Comment:
This would probably be better as a simple object verb, and put in the same file as the detective guns.  The bonus is that once you reskin your gun, you can remove the verb from the weapon so you can't do it again, as verbs are actually contained in a list named verbs.